packages/flutter/lib/src/material/page.dart

```diff
@@ -151,7 +151,12 @@ class MaterialPageRoute<T> extends PageRoute<T> {
 
   @override
   Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
-    final Widget result = builder(context);
+    final Widget result = new Semantics(
+      route: true,
+      explicitChildNodes: true,
+      value: settings.accessibilityName,
+      child: builder(context),
+    );
```

Comment:
I could see putting the name here, though we'd need to put a constructor argument to configure it, and that would work poorly with things like the route table (MaterialApp.routes).

Other options we could consider (maybe do all of them):
 - We could have a `RouteName` widget, similar to `Title`, whose purpose is to populate the semantics tree with a route name.
 - We could have the AppBar automatically do this, e.g. label the contents of the appbar as "isRoute" (if I understand how this is configured on the engine side that would just work, right? cc @goderbauer).
- We could just tell people to use a Semantics widget like we do for dialogs.

Reply:
I think that at this point a minimal first pass seems to be the best thing to land.  I'll create a routeName widget and we can tell people to use semantics for the rest.  If there are issues, we can always add more defaults later - easier than trying to clean them up.

Labeling these as routes will work, a child semantics that is also labeled as a route with a value will get read by the engine implementations instead.